tokio-trace-futures/src/lib.rs

```diff
@@ -44,46 +42,29 @@ pub struct WithDispatch<T> {
 
 impl<T: Sized> Instrument for T {}
 
-impl<'a, T: Future> Future for Instrumented<'a, T> {
+impl<T: Future> Future for Instrumented<T> {
     type Item = T::Item;
     type Error = T::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         let span = &mut self.span;
         let inner = &mut self.inner;
-        match span.enter(|| inner.poll()) {
-            Ok(Async::NotReady) => Ok(Async::NotReady),
-            done => {
-                // The future finished, either successfully or with an error.
-                // The span should now close.
-                span.close();
-                done
-            }
-        }
+        span.enter(|| inner.poll())
```

Comment:
is there no more need to call close?

Reply:
`close` no longer exists. :)

The previous code here was an artifact of a much earlier model of span closure...it still worked against the current `tokio-trace` but was unnecessary.